StudentInsertionClass.cls

```diff
@@ -0,0 +1,26 @@
+public class StudentInsertionClass {
+
+    public static void NotAllowingInsertion(List<Student__c> studentsList){
+        
+       set<Id> classIds = new Set<ID>();
+       for(Student__c student : studentsList) {
+           classIds.add(student.class_object__c);
+       }
+        
+       List<Class__c> classList = [select NoOfStudents__c, MaxSize__c from Class__c where Id IN :classIds];
+       Map<Id,Class__c> classMap = new Map<Id,Class__c>();

```

Comment:
can we get this kind of map directly from query

Reply:
We cant get this map directly from query because from query we got the list of classes in which students want to insert themselves.